### ISSUE TITLE
Schema.name not respected when the class is used to parameterize a generic class

### ISSUE THREAD

**Issue by stiancognite** [swagger-api/swagger-core#2522]
I have this `SomethingDTO`:
```
@Schema(name = "Something", description = "42 test 42")
public class SomethingDTO {
  @Schema(description = "Unique ID")
  private String id;

  public SomethingDTO(Something something) {
    id = something.getId();
  }

  public String getId() {
    return id;
  }
}
```
And I use `@Schema(name = "Something")` because I want the class to be called `Something` in the resulting `openapi.json`. But this is what I get:
```
            "SomethingDTO": {
                "type": "object",
                "properties": {
                    "id": {
                        "type": "string",
                        "description": "Unique ID"
                    }
                },
                "description": "42 test 42"
            }
```
I think this is happening because of how I use `SomethingDTO` (to parameterize a generic type, [see the example code in this previous issue](https://github.com/swagger-api/swagger-core/issues/2516)). Is there something I can do to make the resulting class in `openapi.json` be called `Something` instead of `SomethingDTO`?

I'm using `io.swagger.core.v3:swagger-core:jar:2.0.0-SNAPSHOT`.

**Comment by frantuma** [comment]
This is caused by a bug (basically resolving `@Schema.title` instead of `name` and not resolving in generic type) which is fixed in #2523 and  latest `2.0.0-SNAPSHOT`, please update swagger deps and give it a go

**Comment by stiancognite** [comment]
Thank you again! :) Confirmed working.
